types/joi/index.d.ts

```diff
@@ -825,20 +844,26 @@ export interface Terms {
 export interface Rules {
     name: string;
     params?: ObjectSchema | { [key: string]: Schema };
-    setup?: Function;
-    validate?: Function;
+    setup?: (this: AnySchema<AnySchema<Schema>>, description: { [key: string]: any }) => AnySchema<AnySchema<Schema>> | void;
```

Comment:
Why does `setup` return `x | void` but `validate` return `x | undefined`?  Does it convey a meaningful distinction?